In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv('../../data/raw/telco_customer_churn.csv')

In [2]:
print(df.head())

   customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract Pape

In [3]:
print(df.info())

<class 'pandas.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   str    
 1   gender            7043 non-null   str    
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   str    
 4   Dependents        7043 non-null   str    
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   str    
 7   MultipleLines     7043 non-null   str    
 8   InternetService   7043 non-null   str    
 9   OnlineSecurity    7043 non-null   str    
 10  OnlineBackup      7043 non-null   str    
 11  DeviceProtection  7043 non-null   str    
 12  TechSupport       7043 non-null   str    
 13  StreamingTV       7043 non-null   str    
 14  StreamingMovies   7043 non-null   str    
 15  Contract          7043 non-null   str    
 16  PaperlessBilling  7043 non-null   str    
 17  Paymen

In [5]:
df.isnull().sum()

customerID          0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
MonthlyCharges      0
TotalCharges        0
Churn               0
dtype: int64

In [6]:
print("Data duplicate:", df.duplicated().sum())

Data duplicate: 0


## Data Dictionary (Business-Friendly)

- `customerID`: รหัสลูกค้า (unique identifier)
- `gender`: เพศของลูกค้า (Male/Female)
- `SeniorCitizen`: สถานะผู้สูงอายุ (1 = Yes, 0 = No)
- `Partner`: มีคู่สมรส/คู่ชีวิตหรือไม่ (Yes/No)
- `Dependents`: มีผู้ที่ต้องดูแลหรือไม่ (Yes/No)
- `tenure`: จำนวนเดือนที่เป็นลูกค้า
- `PhoneService`: มีบริการโทรศัพท์หรือไม่ (Yes/No)
- `MultipleLines`: มีหลายเบอร์หรือไม่ (Yes/No/No phone service)
- `InternetService`: ประเภทอินเทอร์เน็ต (DSL/Fiber optic/No)
- `OnlineSecurity`: มีบริการความปลอดภัยออนไลน์หรือไม่
- `OnlineBackup`: มีบริการสำรองข้อมูลออนไลน์หรือไม่
- `DeviceProtection`: มีบริการคุ้มครองอุปกรณ์หรือไม่
- `TechSupport`: มีบริการช่วยเหลือทางเทคนิคหรือไม่
- `StreamingTV`: มีบริการดูทีวีสตรีมมิงหรือไม่
- `StreamingMovies`: มีบริการดูภาพยนตร์สตรีมมิงหรือไม่
- `Contract`: ประเภทสัญญา (Month-to-month/One year/Two year)
- `PaperlessBilling`: ใช้ใบแจ้งหนี้อิเล็กทรอนิกส์หรือไม่
- `PaymentMethod`: วิธีชำระเงินของลูกค้า
- `MonthlyCharges`: ค่าใช้บริการรายเดือน
- `TotalCharges`: ค่าใช้บริการสะสม
- `Churn`: สถานะการยกเลิกบริการ (Yes/No)

หมายเหตุ: นิยามตัวแปรนี้ใช้เพื่อ align ความเข้าใจระหว่างทีมธุรกิจและทีมวิเคราะห์.

In [8]:
#Count churn amount
churn_counts = df['Churn'].value_counts()
print("\nChurn Value Counts:")
print(churn_counts)


Churn Value Counts:
Churn
No     5174
Yes    1869
Name: count, dtype: int64


In [9]:
# Check churn distributione
churn_imbalance_ratio = churn_counts['No'] / churn_counts['Yes']
print(f"\nChurn Imbalance Ratio : {churn_imbalance_ratio:.2f}")

# Check churn distribution to percentage
churn_rate = churn_counts['Yes'] / churn_counts.sum() * 100
print(f"Churn Rate : {churn_rate:.2f}%")


Churn Imbalance Ratio : 2.77
Churn Rate : 26.54%


### Overview Conclusion

- Dataset มีลูกค้า **7,043 ราย** และ churn rate ประมาณ **26.5%**
- ไม่พบ duplicate row ในการตรวจเบื้องต้น
- ข้อมูลพร้อมสำหรับขั้นตอน **Data Quality** และ **Data Preparation**

**Next step:** เปิด [02_data_quality.ipynb](02_data_quality.ipynb) เพื่อทำ quality validation อย่างเป็นระบบ.